sdk/src/beacon.cpp

```diff
@@ -232,7 +232,7 @@ appd_iot_error_code_t appd_iot_send_all_beacons(void)
   }
 
   char jsonlen_buf[10];
-  snprintf(jsonlen_buf, sizeof(jsonlen_buf), "%lu", jsondata.length());
+  snprintf(jsonlen_buf, sizeof(jsonlen_buf), "%lu", (unsigned long)jsondata.length());
```

Comment:
I thought int64_t or int32_t is better and platform independent. But since they are not used in other places, we don't need to use them in the future.